pandas/core/nanops.py

```diff
@@ -212,6 +213,10 @@ def _get_values(values, skipna, fill_value=None, fill_value_typ=None,
             mask = isna(values)
 
     dtype = values.dtype
+    if is_datetime64tz_dtype(orig_values):
```

Comment:
this is a very very odd place to do this as no other dtype specific things are here

Reply:
Semi-agree.  Most of `nanops` is numpy dtype/ndarray specific, so making datetime64tz work is klunky.  That's why I thought it made more sense to implement these directly in DTA in #23890.  But conditional on implementing this here in nanops, this check is necessary.

Reply:
my point is this is in the wrong place. nanops already handles dtypes just not here.

Reply:
so this should go in _view_if_needed (or maybe just remove that function and inline it here is ok

Reply:
did you do this?

Reply:
Haven't yet.  view_if_needed is called ~25 lines below this.  It isn't obvious to me that it is OK to move it up to here.  If it is, then sure, its a one-liner that can be inlined.

Reply:
yes move it, as I said above you have dtype inference in 2 places for no reason.